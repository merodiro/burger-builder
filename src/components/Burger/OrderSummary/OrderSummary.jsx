import React from 'react'
import PropTypes from 'prop-types'

import classes from './OrderSummary.module.css'
import Button from '../../UI/Button/Button'

const OrderSummary = ({
  ingredients,
  price,
  purchaseContinue,
  purchaseCancel,
}) => (
  <>
    <h3>Your Order</h3>
    <p>A delicious burger with the following ingredients:</p>
    <ul>
      {Object.entries(ingredients).map(([key, amount]) => (
        <li key={key}>
          <span className={classes.IngredientKey}>{key}:</span> {amount}
        </li>
      ))}
    </ul>
    <p>
      <strong>Total Price: {price.toFixed(2)}</strong>
    </p>
    <p>Continue to Checkout?</p>
    <Button type="Danger" clicked={purchaseCancel}>
      Cancel
    </Button>
    <Button type="Success" clicked={purchaseContinue}>
      Continue
    </Button>
  </>
)

OrderSummary.prototype = {
  ingredients: PropTypes.object.isRequired,
  purchaseContinue: PropTypes.func.isRequired,
  purchaseCancel: PropTypes.func.isRequired,
  price: PropTypes.number.isRequired,
}

export default OrderSummary
