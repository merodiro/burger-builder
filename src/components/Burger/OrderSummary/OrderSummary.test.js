import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'

import OrderSummary from './OrderSummary'

describe('<OrderSummary>', () => {
  it('renders the same output', () => {
    const ingredients = {
      salad: 2,
      bacon: 1,
      cheese: 1,
      meat: 2,
    }
    const component = shallow(<OrderSummary ingredients={ingredients} />)
    expect(component).toMatchSnapshot()
  })
})
