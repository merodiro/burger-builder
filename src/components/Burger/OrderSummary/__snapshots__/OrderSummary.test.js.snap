// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderSummary> renders the same output 1`] = `
<React.Fragment>
  <h3>
    Your Order
  </h3>
  <p>
    A delicious burger with the following ingredients:
  </p>
  <ul>
    <li
      key="salad"
    >
      <span
        className="IngredientKey"
      >
        salad
        :
      </span>
       
      2
    </li>
    <li
      key="bacon"
    >
      <span
        className="IngredientKey"
      >
        bacon
        :
      </span>
       
      1
    </li>
    <li
      key="cheese"
    >
      <span
        className="IngredientKey"
      >
        cheese
        :
      </span>
       
      1
    </li>
    <li
      key="meat"
    >
      <span
        className="IngredientKey"
      >
        meat
        :
      </span>
       
      2
    </li>
  </ul>
  <p>
    <strong>
      Total Price: 
      5.56
    </strong>
  </p>
  <p>
    Continue to Checkout?
  </p>
  <Button
    type="Danger"
  >
    Cancel
  </Button>
  <Button
    type="Success"
  >
    Continue
  </Button>
</React.Fragment>
`;
